src/app/tasks/task.service.ts

```diff
@@ -14,14 +18,30 @@ export class TaskService {
 
         let taskArray: Task[] = [];
 
-        // JSON data needs to be parsed because of complex objects like Date
-        taskData.forEach(taskElementData => {
+        // Load task types
+        taskData.taskTypes.forEach(taskType => {
+            this.taskTypes[taskType.value] = {
```

Comment:
why use OptionIndex? You can just create an array of Option[] instead of indexing it, so taskTypes: Option[] = [] same for statuses, then you can just assign taskData.taskTypes to this.taskTypes

Reply:
Because in TaskService I had to assign a specific TaskType/TaskStatus to the task, therefore I had "keys" to index the list of options.

Reply:
You can use assign a specific type/status to a task without using indexing